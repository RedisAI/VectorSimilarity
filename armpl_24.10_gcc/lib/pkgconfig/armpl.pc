armpl-static-lp64-seq.pc