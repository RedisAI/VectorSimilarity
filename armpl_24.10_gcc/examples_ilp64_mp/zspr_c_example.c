/*
 * Double complex symmetric rank 1 operation in packed format
 *
 * Arm Performance Libraries version 24.10
 * SPDX-FileCopyrightText: Copyright 2015-2024 Arm Limited and/or its affiliates
 */

#include <stdio.h>
#include <stdlib.h>
#include "armpl.h"

#define N 4

int main()
{
	const char uplo = 'U';

	const int64_t n = N;
	const int64_t incx = 1;

	const double complex alpha = 1.1 + 1.2 * I;

	double complex ap[N*(N+1)/2], x[N];

	printf("ARMPL example: double complex symmetric rank 1 operation using ZSPR\n");
	printf("-------------------------------------------------------------------\n");
	printf("\n");

	/* Initialize matrix A in packed format */
	for (int64_t i = 0; i < n*(n+1)/2; ++i) {
		ap[i] = (double)(i+1) + (double)(i+2) * I;
	}

	/* Initialize vector x */
	for (int64_t i = 0; i < n; ++i) {
		x[i] = (double)(2*i+1) + (double)(2*i+2) * I;
	}

	printf("alpha : %5.3F%+5.3Fi\n", creal(alpha), cimag(alpha));
	printf("\n");

	printf("Matrix A:\n");
	for (int64_t i = 0; i < n; ++i) {
		for (int64_t j = 0; j < n; ++j) {
			int64_t idx = j < i ? j + i*(i+1)/2 : i + j*(j+1)/2;
			printf(" %10.3E%+10.3Ei", creal(ap[idx]), cimag(ap[idx]));
		}
		printf("\n");
	}

	printf("\n");
	printf("Vector x:\n");
	for (int64_t i = 0; i < n; ++i) {
		printf(" %10.3E%+10.3Ei\n", creal(x[i]), cimag(x[i]));
	}

	zspr_(&uplo, &n, &alpha, &x[0], &incx, &ap[0]);

	printf("\n");
	printf("Updated matrix A = alpha*x*x**T + A:\n");
	for (int64_t i = 0; i < n; ++i) {
		for (int64_t j = 0; j < n; ++j) {
			int64_t idx = j < i ? j + i*(i+1)/2 : i + j*(j+1)/2;
			printf(" %10.3E%+10.3Ei", creal(ap[idx]), cimag(ap[idx]));
		}
		printf("\n");
	}

	return 0;
}
