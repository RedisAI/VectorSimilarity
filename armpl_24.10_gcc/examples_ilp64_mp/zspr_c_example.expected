ARMPL example: double complex symmetric rank 1 operation using ZSPR
-------------------------------------------------------------------

alpha : 1.100+1.200i

Matrix A:
  1.000E+00+2.000E+00i  2.000E+00+3.000E+00i  4.000E+00+5.000E+00i  7.000E+00+8.000E+00i
  2.000E+00+3.000E+00i  3.000E+00+4.000E+00i  5.000E+00+6.000E+00i  8.000E+00+9.000E+00i
  4.000E+00+5.000E+00i  5.000E+00+6.000E+00i  6.000E+00+7.000E+00i  9.000E+00+1.000E+01i
  7.000E+00+8.000E+00i  8.000E+00+9.000E+00i  9.000E+00+1.000E+01i  1.000E+01+1.100E+01i

Vector x:
  1.000E+00+2.000E+00i
  3.000E+00+4.000E+00i
  5.000E+00+6.000E+00i
  7.000E+00+8.000E+00i

Updated matrix A = alpha*x*x**T + A:
 -7.100E+00+2.800E+00i -1.550E+01+8.000E+00i -2.290E+01+1.420E+01i -2.930E+01+2.140E+01i
 -1.550E+01+8.000E+00i -3.350E+01+2.200E+01i -5.050E+01+3.700E+01i -6.650E+01+5.300E+01i
 -2.290E+01+1.420E+01i -5.050E+01+3.700E+01i -7.810E+01+5.980E+01i -1.037E+02+8.460E+01i
 -2.930E+01+2.140E+01i -6.650E+01+5.300E+01i -1.037E+02+8.460E+01i -1.409E+02+1.162E+02i
